### PR TITLE
analytics: instrument tracking

### PR DESCRIPTION
**Pull request by pnavarrc**
## Read More
- [Analytics Help: About Events](https://support.google.com/analytics/answer/1033068#)
- [Measure Google Analytics Events](https://developers.google.com/analytics/devguides/collection/gtagjs/events)

### CODE REVIEW

File: src/components/AppBar/AppBar.tsx [hunk 1]
```diff
@@ -138,9 +139,7 @@ const _AppBar = () => {
   const shareTitle = locationName ? locationShareTitle : defaultShareTitle;
 
   const trackShare = (target: string) => {
-    window.gtag('event', 'share', {
-      event_label: target,
-    });
+    trackEvent(EventCategory.ENGAGEMENT, EventAction.SHARE, target);
```
Comment (pnavarrc) [inline src/components/AppBar/AppBar.tsx:142]: Note: `engagement` is the default category for the `share` event, see https://developers.google.com/analytics/devguides/collection/gtagjs/events

File: src/common/utils/tracking.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  COPY_LINK = 'copy link',
+}
+
+export default function trackEvent(
+  category: EventCategory,
+  action: EventAction,
+  label: string,
+  value?: number,
+) {
+  window.gtag('event', action, {
```
Comment (ghop02) [inline src/common/utils/tracking.ts:19]: Would it be worth using the official react GA plugin? https://github.com/react-ga/react-ga

Comment (pnavarrc) [reply src/common/utils/tracking.ts:22]: Oh that's pretty neat, thanks, will give it a try!

Comment (pnavarrc) [reply src/common/utils/tracking.ts:42]: actually I will do that later - I want to debug the issues that Igor is mentioning (page views might be wrong) without the extra layer

**Review by chasulin — APPROVED** [review]

**Pull request merged**